From 2ea642624389265e96c0fb8bf6fd84c93f0f9691 Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Fri, 6 Dec 2024 23:55:35 +0000
Subject: [PATCH] servicemanager: set consistent flags for self-reg

SM has special flags set to become a context object, but
these were not set when it registers as itself.

Bug: 382775095
Test: boot w/ harsher checks
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:0712dc9870ace727ee09dc67bec84d4d3cf41e6d)
Merged-In: I0fb567cbcca67a2fc6c088f652c8af570b8d7e53
Change-Id: I0fb567cbcca67a2fc6c088f652c8af570b8d7e53
---
 cmds/servicemanager/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmds/servicemanager/main.cpp b/cmds/servicemanager/main.cpp
index 2fb9c2bc9a..dac6d96347 100644
--- a/cmds/servicemanager/main.cpp
+++ b/cmds/servicemanager/main.cpp
@@ -126,6 +126,7 @@ int main(int argc, char** argv) {
     ps->setCallRestriction(ProcessState::CallRestriction::FATAL_IF_NOT_ONEWAY);
 
     sp<ServiceManager> manager = sp<ServiceManager>::make(std::make_unique<Access>());
+    manager->setRequestingSid(true);
     if (!manager->addService("manager", manager, false /*allowIsolated*/, IServiceManager::DUMP_FLAG_PRIORITY_DEFAULT).isOk()) {
         LOG(ERROR) << "Could not self register servicemanager";
     }
-- 
2.48.1.262.g85cc9f2d1e-goog

